ta/crypt/cryp_taf.c

```diff
@@ -34,7 +34,10 @@ do {                                             \
 		return TEE_ERROR_BAD_PARAMETERS; \
 } while (0)
 
-#define VAL2HANDLE(v) (void *)(uintptr_t)(v)
+/* pointers in 32-bit fields passed as an offset to ta_head */
+extern const void *ta_head;
+#define VAL2HANDLE(v) (void *)(uintptr_t)(v == TEE_HANDLE_NULL ? v : v + (uintptr_t)&ta_head)
+#define HANDLE2VAL(h) (h == TEE_HANDLE_NULL ? (uintptr_t)h : (uintptr_t)((uintptr_t)h - (uintptr_t)&ta_head))
```

Comment:
since returning a `uint32_t`, i think it should be:
`#define HANDLE2VAL(h) (uint32_t)(h == TEE_HANDLE_NULL ? h : (uintptr_t)h - (uintptr_t)&ta_head)`